### ISSUE TITLE
[Misc]: test

### ISSUE THREAD

**Issue by wangxiyuan** [vllm-project/vllm-ascend#6787]
### Anything you want to discuss about vllm on ascend.

# Release Notes

## v0.15.0rc1 - 2026.02.24

This is the release candidate for v0.15.0 of vLLM Ascend. This release brings significant enhancements in model support, graph optimization, quantization capabilities, and stability improvements across Ascend hardware platforms.

> ⚠️ **Release Candidate Notice**: This is a pre-release version. Please test thoroughly before using in production environments.

---

### Highlights

**Model Support**

* **GLM5**: Added full adaptation support for GLM5 models with deployment guides and A2 series documentation. #6642 #6709 #6711 #6717
* **Qwen3-VL-MoE**: [Experimental] Added EAGLE speculative decoding support for Qwen3-VL-MoE models. #6327
* **Qwen3-Next**: Enhanced MTP + Full Decode support with improved nightly CI stability. #6047
* **DeepSeek-V3.2**: Added long/short prompt test coverage and optimized KV cache usage for better performance. #6499 #6610
* **GLM4.7-Flash**: Added W8A8 quantization support. #6492
* **310P Series**: Expanded support for W8A8 quantization, MoE models, and MoE W8A8 quantization on Ascend 310P hardware. #6454 #6530 #6641

**Core Features**

* **npugraph_ex**: Enabled by default with enhanced graph fusion capabilities including MatmulAllReduceAddRMSNorm and RMSNorm dynamic quant fusion passes. Static kernel installation and e2e test coverage improved. #6664 #6006 #6274 #6320 #6388 #6430 #6513 #6553
* **Context Parallel (PCP/DCP)**: Enhanced support with Mooncake layerwise integration, fixed slot_mapping issues for mixed P/D batches, and unified PD request discrimination logic. #6627 #6344 #6317 #5939
* **EPLB (Elastic Prefill Load Balancing)**: Added fp/bf16 support and fixed dynamic EPLB effectiveness issues. #5531 #6653
* **KV Pool & Mooncake**: Added sparse attention support, remote PTP size retrieval, and improved layerwise connector with recompute scheduler support. #6339 #5822 #5900
* **Speculative Decoding**: Eagle support improvements with dummy_run fixes, target logits correction, and accuracy fixes for PCP/DCP scenarios. #6517 #6685 #6491
* **PD Disaggregation**: Fixed layerwise request_id errors and added production safeguards for 300I. #6360 #6343

---

### Features

* **Quantization**: 
  - Added compressed tensors MoE W4A8 dynamic weight support. #5889
  - Enhanced W8A8 support across multiple model architectures.
* **Memory Management**: 
  - KV pool sparse attention support for more efficient memory utilization. #6339
  - Fixed KV cache leak on P-side due to incorrect req_id. #6325
* **Operator Enhancements**:
  - Added AscendC fused op `transpose_kv_cache_by_block` to accelerate GQA transfer. #6366
  - DispatchGmmCombineDecode now supports bf16/float16 GMM1/GMM2 weights with ND format. #6393
  - Batch invariant implementation with AscendC for improved kernel efficiency. #6590
* **Profiling & Debugging**: Added CUDA graph stats to execution state and Triton info collection in `collect_env.py`. #6331 #6298

---

### Hardware and Operator Support

* **310P/300I DUO Optimization**:
  - Refactored KV cache allocator and attention implementation aligned with main branch. #6270 #6579
  - Added AddRMSNorm and WeightNZ feature support (quant/unquant). #6704 #6705
  - Comprehensive op refactoring with added unit tests. #6296 #6591 #6539
* **npugraph_ex Enhancements**:
  - Static kernel uninstall fix and duplicate pattern resolution. #6128 #6513
  - Extra validation for AllReduce RMSNorm fusion pass. #6430
* **Custom Operators**:
  - Optimized `DispatchFFNCombine` kernel performance and fixed vector alignment issues. #6468 #6707
  - Fixed `rope_forward_triton` errors and enhanced rotary embedding with partial RoPE support. #6404 #6581
  - Added `apply_top_k_top_p_custom` kernel e2e tests. #6348
* **CANN 8.5 Readiness**: Updated UT CANN version to 8.5.0 and enabled previously skipped test cases. #6564 #6195

---

### Performance

* **Kernel Optimizations**:
  - Adaptive block size selection in `linear_persistent` kernel for better throughput. #6537
  - Triton stability improvements for large grid configurations on Ascend. #6301
  - Rope embedding optimization via Triton kernel for significant speedup. #5918 (backported)
* **Memory & Scheduling**:
  - Optimized DeepSeek v3.2 KV cache usage patterns. #6610
  - Removed unnecessary graph size re-initialization and redundant D2H operations. #6280 #4063
  - Fixed sync bugs in `DispatchFFNCombine` to avoid device synchronization overhead. #6416 #6432
* **MoE & Attention**:
  - Enhanced All2AllV for MoE models and refined expert routing logic. #3738 (continuation)
  - Improved attention operator with FIA/TND constraint handling. #6357 #6475

---

### Dependencies

* **vLLM Core**: Upgraded to v0.15.0 main branch. #6470 #6646 #6560 #6673
* **Transformers**: Upgraded to latest compatible version for model compatibility. #6307
* **Mooncake**: Bumped to v0.3.8.post1 for improved KV offloading. #6428
* **Tooling**: 
  - Updated doctest from 0.9.1 to 0.13.0 with nightly CI integration. #6452
  - Enhanced linting with ruff format batches (#7-#11) and Windows adaptation. #6023 #6129 #6173 #6176 #6604 #6727

---

### Deprecation & Breaking Changes

* **Environment Variables**: 
  - Removed unused `USE_OPTIMIZED_MODEL` environment variable. #6618
* **Code Cleanup**:
  - Removed custom `rotary_embedding` operator in favor of upstream Triton implementation. #6523
  - Cleaned up `ProfileExecuteDuration` profiling utility. #6461
  - Dropped DeepSeek-specific patches now handled upstream. #6288
* **API Changes**:
  - `dump_config` in `additional_config` renamed to `dump_config_path` (type changed from `dict` to `string`). #5296 (carried forward)
  - Deprecated `–task` parameter for embedding models. #5257 (carried forward)

---

### Documentation

* **New Guides**:
  - Added `npugraph_ex` introduction and usage documentation. #6306
  - Memcache usage guide for KV offloading scenarios. #6476
  - Benchmark contribution tutorial for suffix speculative decoding. #6323
* **Model Documentation**:
  - GLM5 deployment guide with A2 series hardware instructions. #6709 #6711 #6717
  - Updated reranker guide with deprecated task option removal. #6385
* **Release Management**: Backported v0.13.0 release notes and restructured tutorial documentation. #6584 #6501

---

### Bug Fixes & Stability

* **Critical Fixes**:
  - Fixed MoE load accumulation error in ACL graph mode. #6182
  - Resolved KV cache leak on P-side due to incorrect `req_id` tracking. #6325
  - Fixed intermittent `kv_port` conflicts with `AscendDirectTransport`. #6455
  - Corrected hash conflicts from incompatible configuration resets. #6368
* **Model-Specific Fixes**:
  - Fixed MTP test failures by adding missing `draft_attn_metadatas` parameter. #6232
  - Resolved accuracy issues in PCP/DCP with speculative decoding. #6491
  - Fixed `AddRMSNormQuant` not taking effect in quantized scenarios. #6620
* **CI/CD Improvements**:
  - Fixed wheel package build workflow and nightly image build tagging. #6276 #6333 #6367 #6420
  - Added auto-partitioning for test cases and production safeguards for 300I. #6379 #6343
  - Resolved Qwen3-Next and DeepSeek-R1 nightly CI failures. #6247 #6297

---

### Known Issues

* **Transformers Compatibility**: Some quantized model weights (e.g., `qwen2.5vl`, `gemma3`) may encounter issues due to recent `transformers` package upgrades. Workaround: Pin to compatible version or await patch release. #6302
* **310P Memory Constraints**: MoE models with W8A8 quantization on 310P may require careful memory tuning for large batch sizes. Monitor `VLLM_ASCEND_MEM_POOL_SIZE`.
* **npugraph_ex Experimental**: While enabled by default, some fusion passes may require manual validation for custom model architectures. Report issues via GitHub.

---

### Upgrade Guide

1. **Docker Users**: Pull the latest v0.15.0rc1 image from the official registry.
2. **Source Build**: 
   ```bash
   git checkout releases/v0.15.0rc1
   pip install -e .
   ```
3. **Configuration**: Review deprecated environment variables and update `additional_config` parameters as noted in Breaking Changes.
4. **Testing**: Run model-specific validation suites, especially for quantized MoE and speculative decoding workflows.

> 📬 **Feedback Welcome**: As a release candidate, your testing feedback is invaluable. Please report issues at [vllm-project/vllm-ascend Issues](https://github.com/vllm-project/vllm-ascend/issues).

---

*For detailed migration guidance, refer to the [vLLM Ascend Documentation](https://docs.vllm.ai/projects/ascend/).*
